src/components/Asset/AssetActions/Trade/index.tsx

```diff
@@ -1,19 +1,106 @@
-import React, { ReactElement, useEffect, useState } from 'react'
+import React, { ReactElement, useCallback, useEffect, useState } from 'react'
 import FormTrade from './FormTrade'
 import { useAsset } from '@context/Asset'
 import { useWeb3 } from '@context/Web3'
+import { isValidNumber } from '@utils/numbers'
 import Decimal from 'decimal.js'
-import { Datatoken } from '@oceanprotocol/lib'
+import { Datatoken, LoggerInstance, Pool } from '@oceanprotocol/lib'
+import { getPoolData } from '@utils/subgraph'
+import { PoolData_poolData as PoolDataPoolData } from 'src/@types/subgraph/PoolData'
 
 Decimal.set({ toExpNeg: -18, precision: 18, rounding: 1 })
 
+interface PoolInfo {
+  poolFee: string
+  weightBaseToken?: string
+  weightDt?: string
+  datatokenSymbol: string
+  datatokenAddress: string
+  baseTokenSymbol: string
+  baseTokenAddress: string
+  totalPoolTokens: string
+  totalLiquidityInOcean: Decimal
+}
+
 export default function Trade(): ReactElement {
   const { accountId, balance, web3 } = useWeb3()
   const { isAssetNetwork } = useAsset()
   const [tokenBalance, setTokenBalance] = useState<PoolBalance>()
-  const { asset } = useAsset()
+  const { asset, owner, refreshInterval } = useAsset()
+  const initialPoolInfo: Partial<PoolInfo> = {
+    totalLiquidityInOcean: new Decimal(0)
+  }
+  const [poolInfo, setPoolInfo] = useState<PoolInfo>(
+    initialPoolInfo as PoolInfo
+  )
+  const [poolData, setPoolData] = useState<PoolDataPoolData>()
+  const [fetchInterval, setFetchInterval] = useState<NodeJS.Timeout>()
+
   const [maxDt, setMaxDt] = useState('0')
-  const [maxOcean, setMaxOcean] = useState('0')
+  const [maxBaseToken, setMaxBaseToken] = useState('0')
+
+  const fetchPoolData = useCallback(async () => {
```

Comment:
duplicating fetching logic just to share data between 2 components should be avoided at all times and if that is needed moving logic either to parent component for both child components, or a fetching context provider usually is the best solution. So I did the latter over in #1057 so in here you can do after its merge:

```js
const {
  poolData,
  poolInfo,
  poolInfoUser,
  poolInfoOwner,
  poolSnapshots,
  hasUserAddedLiquidity,
  isRemoveDisabled,
  refreshInterval,
  fetchAllData
} = usePool()
```

And whatever other data you need you would then update the pool context provider.